packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -132,4 +136,9 @@ export class Span implements types.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  private static _now(): number {
+    // performance.now() is relative from process start.
+    return performance.timeOrigin + performance.now();
```

Comment:
What's the motivation for adding `timeOrigin` here?

I personally would prefer to have all the timestamps associated with a span, events, etc. all have monotonic times with `performance.now` and not add `performance.timeOrigin` until export time.

My reasoning:
- There will be a loss in timestamp accuracy by adding the time origin right away instead of deferring that to the exporter to do in a more careful way. Specifically, if a process has been running for < 90 days there is nanosecond accuracy in a `performance.now()` timestamp, but I think only ~millisecond accuracy once you add `peformance.timeOrigin`. We can preserve the differential nanosecond accuracy between events and spans from a given process by more carefully combining `performance.timeOrigin` and a `performance.now` timestamp into an ISO date by combining the fractional and whole number parts together separately. See this [OpenCensus Web utility for that](https://github.com/census-instrumentation/opencensus-web/blob/82464ed231ad829a7a8a72eab17e95577aa813d0/packages/opencensus-web-core/src/common/time-util.ts#L37)
- Also, if we make the timestamps based on `performance.now` it is easier to create spans from browser Performance timing API events, which have a bunch of different timings all using the `performance.now` clock, so not having to add `timeOrigin` for each of them saves a step.

See https://github.com/open-telemetry/opentelemetry-js/issues/19 for more discussion of this

That said, I'm open to adding in the origin time but wanted to make a case for not doing it until export time (and then only carefully to preserve differential nanosecond accuracy of span events and start/end times)

Reply:
I'm fine adding before export but would prefer doing it outside of exporters to avoid logic duplication and mistakes around it. Maybe it could be in the `ReadableSpan` converter. 

Also shouldn't we rename in this case Span's `startTime` and `endTime` to reflect that these are relative times. 

If we use relative times, I think the `span.end(endTime?: number)` can be problematic from the ergonomic point of view. As a user, you have to relate your `endTime` coming from somewhere to the process's `performance.now()` and calculate the correct `endTime`. 

(Btw spec says `performance.now()` is millisecond accurate https://developer.mozilla.org/en-US/docs/Web/API/Performance so the extra digits must be a Chrome feature)

Reply:
I was thinking more and `performance.originTime` in my Chrome also has nanosecond accuracy. If after 90 days the browser starts to rounds nanoseconds for `performance.now()` in the calculation of `performance.originTime + performance.now()` we would only loose < 499 nanoseconds from the accuracy in the worst case. As an end-user, it's sounds acceptable to me.

Reply:
The reasoning about when it starts to lose nanosecond accuracy is more about the inherent storage limitations of the `number` type as a 64-bit float.

So you can try doing `performance.timeOrigin` and `performance.timeOrigin + 1e-6` and it will say the same number. In contrast going `let t = performance.now()` and then doing `t` and `t + 1e-6` will show different numbers.

Regarding having it common for the exporters, I like the idea of exporting an ISO time string in the ReadableSpan type, or perhaps we could expose a utility for it as well.

I agree there might be some trickiness with the ergonomics of passing in numbers for the span functions. But in the browser case the ergonomics are actually nicer if it's a performance.now timestamp. Maybe we talk about that in the SIG?